openvdb/points/AttributeArrayString.cc

```diff
@@ -93,54 +136,64 @@ StringMetaInserter::StringMetaInserter(MetaMap& metadata)
 
 void StringMetaInserter::insert(const Name& name)
```

Comment:
The main complexity is finding the next available ids for the strings you want to insert and in both cases (before and after this change) this would be roughly the same, though it's now quicker to search the id container due to the block structure. Obviously providing a container of names would have had a bigger impact on the old changes due to a single reallocation being required, where as calling StringMetaInserter::insert() now for individual names only requires at most single re-allocation - there is also the potential for StringMetaInserter::insert to combine blocks and in doing so erase elements of the block container, but I found that it was actually more efficient to do this in my tests as it can massively speed up subsequent insertions.

Reply:
Fair enough, thanks for the explanation.